app/v1/controllers/validators/CreateAndAmendReliefInvestmentsValidatorFactory.scala

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.validators
+
+import api.controllers.validators.{Validator, ValidatorOps}
+import api.controllers.validators.resolvers._
+import api.models.domain.TaxYear
+import api.models.errors.{DateOfInvestmentFormatError, MtdError, NameFormatError, UniqueInvestmentRefFormatError}
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import cats.implicits._
+import play.api.libs.json.JsValue
+import v1.models.request.createAndAmendReliefInvestments.{
+  CommunityInvestmentItem,
+  CreateAndAmendReliefInvestmentsBody,
+  CreateAndAmendReliefInvestmentsRequestData,
+  EisSubscriptionsItem,
+  SeedEnterpriseInvestmentItem,
+  SocialEnterpriseInvestmentItem,
+  VctSubscriptionsItem
+}
+
+import javax.inject.Singleton
+import scala.annotation.nowarn
+
+@Singleton
+class CreateAndAmendReliefInvestmentsValidatorFactory extends ValidatorOps {
+
+  private val uniqueInvestmentRefRegex = "^[0-9a-zA-Z{À-˿’}\\- _&`():.'^]{1,90}$".r
+
+  private val nameRegex = "^[0-9a-zA-Z{À-˿'}\\- _&`():.'^]{1,105}$".r
+
+  private val resolveParsedNumber = ResolveParsedNumber()
+
+  @nowarn("cat=lint-byname-implicit")
+  private val resolveJson = new ResolveNonEmptyJsonObject[CreateAndAmendReliefInvestmentsBody]()
+
+  private val valid = Valid(())
+
+  def validator(nino: String, taxYear: String, body: JsValue): Validator[CreateAndAmendReliefInvestmentsRequestData] =
+    new Validator[CreateAndAmendReliefInvestmentsRequestData] {
+
+      def validate: Validated[Seq[MtdError], CreateAndAmendReliefInvestmentsRequestData] =
+        (
+          ResolveNino(nino),
+          ResolveTaxYear(TaxYear.minimumTaxYear.year, taxYear, None, None),
+          resolveJson(body)
+        ).mapN(CreateAndAmendReliefInvestmentsRequestData) andThen validateBusinessRules
+
+      private def validateBusinessRules(
+          parsed: CreateAndAmendReliefInvestmentsRequestData): Validated[Seq[MtdError], CreateAndAmendReliefInvestmentsRequestData] = {
+        import parsed.body._
+
```

Comment:
@mattstephens-hmrc for these it could all be done in far fewer lines if the different models inherit from a single trait and then we add the custom reads/writes where necessary. I didn't do it as it felt like scope creep but now i look at it again and I sort of want to update it because it is annoying to see the repetitive use of validateItem where the fields have to be passed in explicitly. What do you think?